chef/cookbooks/ceph/recipes/calamari.rb

```diff
@@ -0,0 +1,65 @@
+node['ceph']['calamari']['packages'].each do |pkg|
+  package pkg
+end
+
+# TODO: do we need pacemaker sync marks? (nothing here uses pacemaker;
+# something for further consideration is making the calamari server HA)
+
+ruby_block "initialize calamari server" do
+  block do
+    check_cmd = Mixlib::ShellOut.new("/srv/www/calamari/manage.py check >/dev/null 2>&1")
+    check_cmd.run_command
+    if check_cmd.exitstatus != 0
+      Chef::Log.info("Ceph: Initializing Calamari")
+      # If check failed, it'll be because calamari isn't initialized yet, so
+      # set it up (newer versions of Mixlib::ShellOut have an error? method
+      # we could call instead of checking exitstatus, but this seems to be
+      # sadly missing on SLE at the moment)
+      init_cmd = Mixlib::ShellOut.new(
+        "calamari-ctl", "initialize",
+          "--admin-username", node[:ceph][:calamari][:username],
+          "--admin-password", node[:ceph][:calamari][:password],
+          "--admin-email", node[:ceph][:calamari][:email])
```

Comment:
I don't think it's actually used ATM. Some digging indicates it's possible to make Django send emails on 500 errors, but AFAICT this isn't actually presently hooked up to the admin email. If it ever *does* get hooked up, then some SMTP config will presumably be necessary, but I don't think we need to do anything about this right now.